### PR TITLE
CB-21851 Readme for secret rotation framework

### PR DESCRIPTION
**Pull request by horadla23**
See detailed description in the commit message.

### CODE REVIEW

**Review by foldik — CHANGES_REQUESTED** [review]
I left a few comments because I found it a bit too technical and hard to grasp.

File: common/src/main/java/com/sequenceiq/cloudbreak/rotation/secret/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+## Core concept
+
+The basic idea is that we have several secrets and we need to replace them with new value at several places.
+Based on that, we would need a list of secrets and we would need to define all places, where the given secret should be replaced.
+Rotation framework has been created based on this idea.
+
+## Technical details
+
+We have enum for secrets/service. The basic interface for the enum that is `SecretType`.
```
[the remark this answers is not in the document]
Comment (horadla23) [reply common/src/main/java/com/sequenceiq/cloudbreak/rotation/secret/README.md:9]: I like this, will replace it with this

File: common/src/main/java/com/sequenceiq/cloudbreak/rotation/secret/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+Every service should have `ApplicationSecretRotationInformation` implemented and that should define an enum for the service which implements the `SecretType` interface.
+Every `SecretType` enum should have the list of all secrets of the service and every secret has an ordered list of `SecretRotationStep`, SecretRotationStep belongs to a place/context (database on provider side, salt pillar, cluster manager user) where we need to replace the secret itself.
+
+### Every SecretRotationStep has 2 corresponding classes
```
[the remark this answers is not in the document]
Comment (horadla23) [reply common/src/main/java/com/sequenceiq/cloudbreak/rotation/secret/README.md:13]: agree

File: common/src/main/java/com/sequenceiq/cloudbreak/rotation/secret/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+- an executor class implementing the `RotationExecutor` interface
+- a context, provided to the executor and used to hold relevant data for the rotation of the given step, extending the `RotationContext` class
+
+### RotationExecutor should implement 3 phases
```
[the remark this answers is not in the document]
Comment (horadla23) [reply common/src/main/java/com/sequenceiq/cloudbreak/rotation/secret/README.md:17]: agree

**Review by foldik — APPROVED** [review]
LGTM

**Pull request merged**